### PR TITLE
K8SPXC-771 - add-configuration for proxysql and haproxy

### PR DESCRIPTION
**Pull request by spron-in**
Add proxysql and haproxy configuration fields.

### CODE REVIEW

**Review by cap1984 — APPROVED** [review]

File: charts/pxc-db/production-values.yaml [hunk 1]
```diff
@@ -93,15 +93,53 @@ pxc:
...
+    timeoutSeconds: 5
+    periodSeconds: 10
+    successThreshold: 1
+    failureThreshold: 3
+
 
 haproxy:
   enabled: true
   size: 3
-  image:
```
Comment (delgod) [inline charts/pxc-db/production-values.yaml:100]: should we keep backward compatibility?

**Review by delgod — APPROVED** [review]

File: charts/pxc-db/production-values.yaml (hunk 1 above)
Comment (cap1984) [reply charts/pxc-db/production-values.yaml:100]: Don't thinks it's necessary

Comment (cap1984) [reply charts/pxc-db/production-values.yaml:100]: @spron-in what do you think?

File: charts/pxc-db/README.md [hunk 2]
```diff
@@ -67,12 +67,22 @@ The chart can be customized using the following configurable parameters:
 | `pxc.persistence.size` | Sets K8S persistent storage size for all PXC Pods | `8Gi`                      |
 | `pxc.disableTLS` | Disable PXC Pod communication with TLS | `false`                      |
 | `pxc.certManager` | Enable this option if you want the operator to request certificates from `cert-manager` | `false`                      |
+| `pxc.readinessProbes.failureThreshold` | When a probe fails, Kubernetes will try failureThreshold times before giving up | `5`                      |
```
Comment (tplavcic) [inline charts/pxc-db/README.md:70]: New options added in `1.9.0` should be added (`pxc.expose` and `pxc.replicationChannels`): https://github.com/percona/percona-xtradb-cluster-operator/blob/main/deploy/cr.yaml#L39-L55

File: charts/pxc-db/production-values.yaml [hunk 3]
```diff
@@ -207,23 +303,19 @@ proxysql:
...
 
 backup:
   enabled: true
-  image:
-    repository: percona/percona-xtradb-cluster-operator
-    tag: 1.8.0-pxc8.0-backup
+  image: ""
   imagePullSecrets: []
   # - name: private-registry-credentials
   pitr:
```
Comment (tplavcic) [inline charts/pxc-db/production-values.yaml:321]: we should add the following password also: `replication: insecure-replication-password` in `secrets.passwords` part.

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: charts/pxc-db/Chart.yaml [hunk 4]
```diff
@@ -1,12 +1,14 @@
 apiVersion: v1
-appVersion: "1.8.0"
+appVersion: 1.9.0
 description: A Helm chart for installing Percona XtraDB Cluster Databases using the PXC Operator.
 name: pxc-db
 home: https://www.percona.com/doc/kubernetes-operator-for-pxc/kubernetes.html
-version: 0.1.18
+version: 0.1.19
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/Chart.yaml]: done

File: charts/pxc-db/crds/crd.yaml [hunk 5]
```diff
@@ -14,66 +14,444 @@ spec:
...
+      - name: Age
+        type: date
+        jsonPath: .metadata.creationTimestamp
+      subresources:
+        status: {}
+        scale:
+          specReplicasPath: .spec.pxc.size
+          statusReplicasPath: .status.pxc.ready
+          labelSelectorPath: .status.pxc.labelSelectorPath
+    - name: v1-10-0
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/crds/crd.yaml]: done

File: charts/pxc-db/templates/cluster.yaml [hunk 6]
```diff
@@ -109,19 +109,28 @@ spec:
...
+  {{- if or (not .Values.haproxy.enabled) .Values.proxysql.enabled }}
   haproxy:
     enabled: false
   {{- else }}
   {{- $haproxy := .Values.haproxy }}
   haproxy:
     enabled: true
     size: {{ $haproxy.size }}
-    image: {{ $haproxy.image.repository }}:{{ $haproxy.image.tag }}
+    image: {{ include "pxc-db.haproxy-image" .}}
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/templates/cluster.yaml]: done

File: charts/pxc-db/templates/cluster.yaml [hunk 7]
```diff
@@ -238,7 +255,7 @@ spec:
   {{- else }}
     {{- $logcollector := .Values.logcollector }}
     enabled: true
-    image: {{ $logcollector.image.repository }}:{{ $logcollector.image.tag }}
+    image: {{ include "pxc-db.logcollector-image" .}}
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/templates/cluster.yaml]: done

File: charts/pxc-db/templates/cluster.yaml [hunk 8]
```diff
@@ -256,7 +273,7 @@ spec:
   {{- if not $backup.enabled }}
     enabled: false
   {{- end }}
-    image: {{ $backup.image.repository }}:{{ $backup.image.tag }}
+    image: {{ include "pxc-db.backup-image" .}}
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/templates/cluster.yaml]: done

File: charts/pxc-db/Chart.yaml [hunk 9]
```diff
@@ -1,12 +1,14 @@
 apiVersion: v1
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/Chart.yaml]: done

File: charts/pxc-db/README.md (hunk 2 above)
Comment (cap1984) [reply charts/pxc-db/README.md]: done

File: charts/pxc-db/production-values.yaml (hunk 3 above)
Comment (cap1984) [reply charts/pxc-db/production-values.yaml:338]: done

File: charts/pxc-db/production-values.yaml [hunk 10]
```diff
@@ -26,7 +26,7 @@ pxc:
   size: 3
   image:
     repository: percona/percona-xtradb-cluster
-    tag: 8.0.22-13.1
+    tag: 8.0.23-14.1
   autoRecovery: true
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/production-values.yaml:30]: done

File: charts/pxc-db/production-values.yaml [hunk 11]
```diff
@@ -13,7 +13,7 @@ finalizers:
 nameOverride: "production"
 fullnameOverride: "production"
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/pxc-db/production-values.yaml:15]: done

**Review by tplavcic — APPROVED** [review]

**Pull request merged**